test/software/amazon/ion/TimestampTest.java

```diff
@@ -860,6 +860,57 @@ public void testNewTimestampFromLong()
         assertEquals("2010-02-01T10:11:12.340Z", ts.toZString());
     }
 
+    /**
+     * Regression test for https://github.com/amzn/ion-java/issues/160
+     */
+    @Test
+    public void testNewTimestampFromLongYearOneRegressionBug()
+    {
+        // This is the same as the private field Timestamp.MINIMUM_ALLOWED_TIMESTAMP_IN_MILLIS
+        final long MINIMUM_TIMESTAMP_MILLIS = -62135769600000L;
+        // This is the minimum timestamp instantiating by parsing a timestamp string,
+        // which is not effected by the `java.util.Date` bug.
+        final Timestamp MINIMUM_TIMESTAMP = Timestamp.valueOf("0001-01-01T00:00:00.000Z");
+        // Save the original timezone since we modify it below.
+        final TimeZone originalTimeZone = TimeZone.getDefault();
+
+        try {
+            // The `java.util.Date` bug has a different range of times depending on the local offset, so
+            // let's test against all time zones.
+            for (final String tzId : TimeZone.getAvailableIDs()) {
+                final TimeZone tz = TimeZone.getTimeZone(tzId);
+                TimeZone.setDefault(tz);
+
+                // The Timestamps under test must be instantiated *after* the the call to TimeZone.setDefault()
+                // since `java.util.Date` references the default TimeZone when calculating the values for its
+                // date field accessor methods (i.e. Date.get*()).
+                Timestamp minimumTimestampFromMillis = Timestamp.forMillis(MINIMUM_TIMESTAMP_MILLIS, 0);
+                assertEquals(MINIMUM_TIMESTAMP, minimumTimestampFromMillis);
+
+                // This will be the latest millisecond in which the bug with `java.util.Date` can happen
+                Timestamp maximumTimestampWithBug =
+                    Timestamp.forMillis(MINIMUM_TIMESTAMP_MILLIS - tz.getRawOffset(), 0);
+                assertEquals(1, maximumTimestampWithBug.getYear());
+
+                // Test one millisecond after, just to be sure.
+                Timestamp timestampPlusOne =
+                    Timestamp.forMillis(MINIMUM_TIMESTAMP_MILLIS - tz.getRawOffset() + 1, 0);
+                assertEquals(1, timestampPlusOne.getYear());
+
+                // Don't attempt to test this last case since it would create a timestamp that is before year 1.
+                if(tz.getRawOffset() >= 0) break;
```

Comment:
I think you mean `continue` instead of `break`.  My preference would be:

```java
if(tz.getRawOffset() < 0) {
    Timestamp timestampMinusOne = ...;
    assertEquals(...);
}
```

Reply:
Indeed.  Will be addressed.